docs/fundamentals/codespaces.md

```diff
@@ -0,0 +1,130 @@
+
+* Automatically configure port forwarding with the correct protocol.
+* Automatically translate URLs in the .NET Aspire dashboard.
+
+Prior to .NET Aspire 9.1 it was still possible to use .NET Aspire within a GitHub Codespace, however more manual configuration was required.
+
+## GitHub Codespaces vs. Devcontainers
+
+GitHub Codespaces is builds upon Visual Studio Code and the Devcontainers spec. In addition to supporting GitHub Codespaces, .NET Aspire 9.1 also improves support for using Visual Studio Code and locally hosted Devcontainers. While the experience is similar between each approach there are some differences and so we have prepared separate documentation if you want to use .NET Aspire with Devcontainers locally. If you want to use .NET Aspire with Devcontainers locally see [.NET Aspire with Devcontainers in Visual Studio Code](devcontainers.md).
+
+## Quick start using template repository
+
+Configuration of GitHub Codespaces is controlled by a a `.devcontainer/devcontainer.json` file in your own repository. The easiest way to get started using .NET Aspire in GitHub Codespaces is to create a new repository using our [template repository](https://github.com/dotnet/aspire-devcontainer) by following the following steps:
+
+1. [Create a new repository](https://github.com/new?template_name=aspire-devcontainer&template_owner=dotnet) using our template.
+
+![Create new repository](media/new-repository-from-template.png)
+
+Once you provide the details and click _Create repository_ the repository will be created and you will be shown the repository in GitHub.
+
+2. From the new repository click on the Code button and select the Codespaces tab and then click _Create codespace on main_.
+
+![Create codespace](media/create-codespace-from-repository.png)
+
+Once you click on this button you will be navigate to a web-based version of Visual Studio Code. Before using the Codespace the containerized development environment needs to be prepared. This process happens automatically on the server and you can review progress by clicking on the _Building codespace_ link in the notification in the bottom right of the browser window.
+
+![Building codespace](media/building-codespace-image.png)
+
+Once the container image has finished being built the _Terminal_ prompt appears which signals that the environment is ready to be interacted with.
+
+![Codespace terminal prompt](media/codespace-terminal.png)
+
+At this point the .NET Aspire templates have been pre-installed and the ASP.NET Core developer certificate has been pre-configured.
+
+3. Create a new .NET Aspire project using the starter template.
+
+```dotnetcli
+dotnet new aspire-starter --name HelloAspire
+```
+
+This will result in a number of files and folders being created in the repository which will be visible in the _Explorer_ panel on the left hand side of the window.
+
+![Codespaces Explorer panel](media/codespaces-explorer-panel.png)
```

Comment:
I cannot make a suggestion for the code block above this, but it should also be indented 4-spaces so that it renders as a child of this numbered list.

```suggestion
    This results in a number of files and folders being created in the repository, which are visible in the **Explorer** panel on the left side of the window.

    :::image source="media/codespaces-explorer-panel.png" lightbox="media/codespaces-explorer-panel.png" alt-text="Codespaces Explorer panel":::
```